### PR TITLE
Mounted Police Officer

### PR DESCRIPTION
**Pull request by Geevies**
#### Summary
Content "Added a 'Mounted Police Officer' profession, which starts with an armored police horse"

#### Purpose of change

There's a couple changes here, so I'm going to break each down into a paragraphs with a heading each.

- **The Profession**
I wanted to add a role for people that want to get into mounted-ish combat quickly from the start of the game. The mounted officer would have been dispatched to quell the riots, but would have probably fled after cohesion within the police force broke down after the 'riots' kicked into high gear, hence why they have one of the very few trained police horses that didn't perish.

- **The Trained Police Horse**
Horses are notorious for getting scared super easily, even when trained. While researching to make this PR, I saw a video of a horse getting scared of a pride flag that got painted on a road, which is funny, but also showcases the police training doesn't eliminate fear completely. In this PR's iteration, the training allows the horse to move closer to the enemies, even within the 10 tiles range, but will still buck the rider off if the %-chance to get scared hits. This means they aren't fearless warhorses, but you still have some lee-way to get into bow-shooting range if you're willing to take the %-chance of falling off.

- **Mounts Spawning with Items**
One thing that annoyed me with the profession pet types, is that you couldn't make them spawn with items attached, so I added a system to let the monster type spawn with the items already attached, which makes more sense to me lorewise. Plus, the horse armor and tack is too big to fit in storage, in only one could be wielded.

#### Describe the solution

- The profession is just a simple profession added via JSON. Survival level 4 to point out that they're a skilled rider. They have a silenced 9mm USP pistol in a large holster to not scare their horse when they fire.
- The trained police horse is just a horse with slightly higher (20 -> 35) morale and the COMBAT_MOUNT flag, which halves the chance to get bucked off when the horse gets scared, and lets the horse willingly move closer to enemies when mounted. The name is unchanged, with only the description pointing out its training. It spawns with a long rope as a leash, horse tack as a saddle, and kevlar armor horse, as it was deployed to deal with rioters.
- Effectively copy pasted the mech_battery code in order to let mountable animals spawn in with leashes, saddles, armor, and bags.
- Copy-pasted the horse riding sprite definition for Ultica and put the horse's ID in, that's the only tileset I'm really familiar with, and I'm not sure if I should be touching them in the first place.

#### Describe alternatives you've considered

- Make the profession spawn with a regular horse. Would've been easier, but also seems boring.
- Make the profession dictate what the pet spawns with instead of assigning the items to spawn on the monster itself. Doable, but outside my skillset at the moment.

#### Testing

- Spawned in as the Mounted Police Officer, made sure the pet horse had the right description.
- Checked that the examine menu showed that I can take off the leash, armor, and tack.
- Took off the leash, armor, and tack and confirmed that the right items were being returned.
- Rode the horse to a nearby location with enemies, confirmed that I could bring the horse close to them.
- Tested keeping 8-9 tiles away and falling off rarely, then moving closer to 3-4 tiles and falling off quite often.

#### Additional context

This is my first time contributing to C:DDA and working with C++, so please let me know if I made a mistake somewhere, or if my copy-pasted code is garbage. I won't mind.

### CODE REVIEW

File: gfx/UltimateCataclysm/tile_config.json [hunk 1]
```diff
@@ -14838,6 +14838,7 @@
       "sprite_offset_y": -27,
       "tiles": [
         { "id": "rid_mon_horse", "fg": [ { "weight": 1, "sprite": 9869 }, { "weight": 1, "sprite": 9870 } ] },
+        { "id": "rid_mon_horse_police", "fg": [ { "weight": 1, "sprite": 9869 }, { "weight": 1, "sprite": 9870 } ] },
```
[the remark this answers is not in the document]
Comment (Geevies) [reply gfx/UltimateCataclysm/tile_config.json]: My bad. I tried using looks_like but that didn't set me at the correct riding position, is there an alternative, or is that just something the automated tileset thing will handle eventually? I'll push the changes regardless.

[the remark this answers is not in the document]
Comment (irwiss) [reply gfx/UltimateCataclysm/tile_config.json]: not sure what's the correct solution, may be poke tileset people in discord if you don't get a hint here

[the remark this answers is not in the document]
Comment (anothersimulacrum) [reply gfx/UltimateCataclysm/tile_config.json]: Tileset data is modified at https://github.com/I-am-Erk/CDDA-Tilesets

[the remark this answers is not in the document]
Comment (Geevies) [reply gfx/UltimateCataclysm/tile_config.json]: Awesome, thanks. Then this one should be good to go/review code-wise.

File: src/monster.cpp [hunk 2]
```diff
@@ -106,6 +105,10 @@ static const efftype_id effect_venom_player2( "venom_player2" );
 static const efftype_id effect_venom_weaken( "venom_weaken" );
 static const efftype_id effect_webbed( "webbed" );
 static const efftype_id effect_worked_on( "worked_on" );
+static const efftype_id effect_leashed( "leashed" );
```
Comment (bombasticSlacks) [inline src/monster.cpp:108]: These need to be alphabetized.

Comment (bombasticSlacks) [reply src/monster.cpp:108]: The clang build that is failing has more details.

**Review by bombasticSlacks — CHANGES_REQUESTED** [review]

**Pull request merged**